lib/matplotlib/axis.py

```diff
@@ -1430,7 +1434,15 @@ def update_units(self, data):
         if *data* is registered for unit conversion.
         """
 
+        if self.converter is not None:
+            _log.debug('Axis already has a converter')
+            return True
```

Comment:
I think this method should only return `True` if a converter is registered **and** *data* can be converted using that converter.